### PR TITLE
Add filters popover on marketplace

### PR DESCRIPTION
**Pull request by remidej**
### What does it do?

Screenshot 2022-10-10 at 14 09 26

Adds a filters popover on the marketplace to select collections and categories for plugins, and just collections for providers. It does not update the results.

I added the selected tab to the query params. I thought now that we start using them for filters & sort, it's a good opportunity to store all the state there, so we can be sure of where a link guides a user.

I refactored some existing tests. Since the state is shared and reused between each test, it didn't make sense to test the default tab at the end.

I updated the snapshot test to account for the filters. Also I tested that the popover could open, but I found that testing the interactivity further was very hard. I can look into it more though if you think that's necessary.

### CODE REVIEW

File: packages/core/admin/admin/src/pages/MarketplacePage/tests/index.test.js [hunk 1]
```diff
@@ -65,19 +71,47 @@ describe('Marketplace page', () => {
...
 
+    const tabPanel = getByRole('tabpanel');
+    const providerCardText = getByText(tabPanel, 'Cloudinary');
+    const pluginCardText = queryByText(tabPanel, 'Comments');
+    const submitProviderText = queryByText(container, 'Submit provider');
+
+    expect(providersTabActive).not.toBe(null);
+    expect(providerCardText).toBeVisible();
+    expect(submitProviderText).toBeVisible();
+    expect(pluginCardText).toEqual(null);
```
Comment (markkaylor) [inline packages/core/admin/admin/src/pages/MarketplacePage/tests/index.test.js:112]: I don't really understand this change for providers and plugins, what's the added benefit? 

I think it's normal to have similar state setups in tests and it is better to test specific things in isolation. Here we test the snapshot, and fireEvent to test the providers tab is set to active, and test the correct content is displayed after that event.

File: packages/core/admin/admin/src/pages/MarketplacePage/tests/index.test.js [hunk 2]
```diff
@@ -90,9 +124,12 @@ describe('Marketplace page', () => {
     expect(trackUsage).toHaveBeenCalledTimes(1);
   });
 
-  it('should return plugin search results matching the query', async () => {
+  it('should return plugin search results matching the query', () => {
     const { container } = render(App);
-    const input = await getByPlaceholderText(container, 'Search');
+    const pluginsTab = screen.getByRole('tab', { name: /plugins/i });
+    fireEvent.click(pluginsTab);
```
Comment (markkaylor) [inline packages/core/admin/admin/src/pages/MarketplacePage/tests/index.test.js:130]: Why is this necessary now?

**Review by markkaylor — COMMENTED** [review]
Nice work!

I find some of these behaviors a bit strange:
- The filters button doesn't have a close on click outside, but the filter selects do
- The filters popover is navigable using tab, but spacebar or down arrow to open the select also moves you down the page
- The filter selects are not navigable or selectable using the keyboard
- The filter button closes when you click the button, the select button does not close when you click the button

File: packages/core/admin/admin/src/pages/MarketplacePage/tests/index.test.js (hunk 1 above)
Comment (remidej) [reply packages/core/admin/admin/src/pages/MarketplacePage/tests/index.test.js:112]: The problem I see is that it's not just similar test setups, our tests literally share the same state. Updating it in one test affects the other. So previously testing the "default" tab only worked because the previous test did a click on the plugins tab.

But it's true the change a made is more of a workaround than anything. I can look into making sure the state is reset between each test

File: packages/core/admin/admin/src/pages/MarketplacePage/tests/index.test.js (hunk 2 above)
Comment (remidej) [reply packages/core/admin/admin/src/pages/MarketplacePage/tests/index.test.js:130]: Related to my comment above, it was about making sure the tests work in isolation, but it's not the right approach

File: packages/core/admin/admin/src/pages/MarketplacePage/tests/index.test.js (hunk 1 above)
Comment (markkaylor) [reply packages/core/admin/admin/src/pages/MarketplacePage/tests/index.test.js:112]: oh ok gotcha, maybe we could organize them into describe suites for plugins and providers and do some beforeEach magic to make sure each test has a clean state (another PR). But works for me thanks for explaining.

File: packages/core/admin/admin/src/pages/MarketplacePage/components/NpmPackagesFilters/FiltersPopover.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  const { formatMessage } = useIntl();
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+  };
+
+  return (
+    <Popover source={source} padding={3} spacing={4} onBlur={() => {}}>
+      <FocusTrap onEscape={onToggle}>
+        <form onSubmit={handleSubmit}>
```
[the remark this answers is not in the document]
Comment (remidej) [reply packages/core/admin/admin/src/pages/MarketplacePage/components/NpmPackagesFilters/FiltersPopover.js:28]: I was initially copying the FilterPopoverURLQuery, and it looks like I forgot to remove this bit

**Review by markkaylor — APPROVED** [review]
Cool, thanks for the changes LGTM
